x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/common/validation/SourceDestValidator.java

```diff
@@ -419,6 +426,46 @@ public void validate(Context context, ActionListener<Context> listener) {
         }
     }
 
+    public static class RemoteClusterMinimumVersionValidation implements SourceDestValidation {
+
+        private final Version minExpectedVersion;
+
+        public RemoteClusterMinimumVersionValidation(Version minExpectedVersion) {
+            this.minExpectedVersion = minExpectedVersion;
+        }
+
+        @Override
+        public void validate(Context context, ActionListener<Context> listener) {
+            Set<String> remoteClusterNames = context.getRemoteClusterService().getRegisteredRemoteClusterNames();
+            Map<String, Version> remoteClusterVersions;
+            try {
+                remoteClusterVersions =
+                    remoteClusterNames.stream()
+                        .collect(toMap(
+                            cluster -> cluster,
+                            cluster -> context.getRemoteClusterService().getConnection(cluster).getVersion()));
```

Comment:
`getConnection` requires that the node that it is calling from has the `remote_cluster_client` role. Put transform is executed on a `master` node and it is conceivable that the master node does not allow remote connections. 

Consequently, this could fail due to the current node not being a remote cluster client. Which could be a false negative if the remote cluster is actually the appropriate version.

Reply:
We should probably do this remote connection version check (or remote node minversion check) when the task gets assigned to a node. Failing the transform task with an error if a remote cluster is of an old version.

This affords a nice, loud warning + a path forward for the user to correct the problem.

Reply:
I was experimenting with the `remote_cluster_client` role a bit, and my conclusion is the following:
- `remote_cluster_client` is required not only to call `getConnection` but also to perform licence check, or even call `GET /_remote/info`. Consequently, this PR is not making things worse, i.e. if the `master` node is missing `remote_cluster_client` role, the `PUT` will fail anyway due to licence check.

In order to properly fix this, we would need to redirect the validation to any `remote_cluster_client` node we can find. That would require either turning the validation into an action (and make it redirected) or making `_preview` serve as a validation action (and again, make it redirected).
In any case, this is a bit more involved and therefore, a `7.13` rather than `7.12` task. I'll followup on this problem in a separate GH issue/doc.